From 7ba1695a64331f1eeb5480f6386939b5f984293f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=BD=D0=B0=D0=B1?= <nabijaczleweli@nabijaczleweli.xyz>
Date: Sat, 6 Nov 2021 18:21:18 +0100
Subject: [PATCH 2/4] remailer.c: format size_t as size_t, not long
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

x32 warning:
remailer.c: In function ‘mix_redraw_head’:
remailer.c:413:60: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 4 has type ‘size_t’ {aka ‘unsigned int’} [-Wformat=]
  413 |   snprintf(buf, sizeof(buf), "-- Remailer chain [Length: %ld]", chain ? chain->cl : 0);
      |                                                          ~~^    ~~~~~~~~~~~~~~~~~~~~~
      |                                                            |                      |
      |                                                            long int               size_t {aka unsigned int}
      |                                                          %d
---
 remailer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/remailer.c b/remailer.c
index 64ce593e2..3d3098244 100644
--- a/remailer.c
+++ b/remailer.c
@@ -410,7 +410,7 @@ static void mix_redraw_chain(struct MuttWindow *win, struct Remailer **type2_lis
 static void mix_redraw_head(struct MuttWindow *win, struct MixChain *chain)
 {
   char buf[1024];
-  snprintf(buf, sizeof(buf), "-- Remailer chain [Length: %ld]", chain ? chain->cl : 0);
+  snprintf(buf, sizeof(buf), "-- Remailer chain [Length: %zu]", chain ? chain->cl : 0);
   sbar_set_title(win, buf);
 }
 
-- 
2.33.1

